File: moon_sdk_app/src/main/java/com/meili/moon/sdk/app/base/page/widget/Menu.java
package com.meili.moon.sdk.app.base.page.widget;

import android.view.View;
import android.view.ViewGroup;

/**
 * menu的基类
 * Created by imuto on 16/5/20.
 */
public abstract class Menu {
    ViewGroup layout;
    View redTips;

    Menu() {
    }

    Menu(ViewGroup layout, View redTips) {
        this.layout = layout;
        this.redTips = redTips;
    }

    public void setVisibility(int visibility) {
        this.layout.setVisibility(visibility);
    }

    public void setEnabled(boolean enabled) {
        this.layout.setEnabled(enabled);
    }

    public void setClickable(boolean clickable) {
        this.layout.setClickable(clickable);
    }

    public void setOnClickListener(View.OnClickListener listener) {
        this.layout.setOnClickListener(listener);
    }

    public void setVisibleRedTips(boolean visible) {
        redTips.setVisibility(visible ? View.VISIBLE : View.GONE);
    }

    public ViewGroup getRootView() {
        return layout;
    }
}

File: moon_sdk_page_router/src/main/java/com/meili/moon/sdk/page/titlebar/Menu.kt
package com.meili.moon.sdk.page.titlebar

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.meili.moon.sdk.util.app
import kotlinx.android.extensions.LayoutContainer

/**
 * menu的基类
 */
abstract class Menu : LayoutContainer {
    val rootView: ViewGroup by lazy {
        return@lazy LayoutInflater.from(app).inflate(getLayoutResId(), null, false) as ViewGroup
    }

    lateinit var redTips: View

    override val containerView: View? = rootView

    abstract fun getLayoutResId(): Int

    fun setVisibility(visibility: Int) {
        this.rootView.visibility = visibility
    }

    fun setEnabled(enabled: Boolean) {
        this.rootView.isEnabled = enabled
    }

    fun setClickable(clickable: Boolean) {
        this.rootView.isClickable = clickable
    }

    fun setOnClickListener(listener: View.OnClickListener) {
        this.rootView.setOnClickListener(listener)
    }

    fun setVisibleRedTips(visible: Boolean) {
        redTips.visibility = if (visible) View.VISIBLE else View.GONE
    }
}
